homeassistant/components/fritz/common.py

```diff
@@ -196,15 +202,15 @@ def __init__(self) -> None:
 class FritzDevice:
     """FritzScanner device."""
 
-    def __init__(self, mac, name=None):
+    def __init__(self, mac: str, name: str) -> None:
         """Initialize device info."""
-        self._mac = mac
-        self._name = name
-        self._ip_address = None
-        self._last_activity = None
-        self._connected = False
+        self._mac: str = mac
+        self._name: str = name
+        self._ip_address: str | None = None
+        self._last_activity: datetime | None = None
+        self._connected: bool = False
```

Comment:
It's better to use inline hints only when type can't be determined automatically.
```suggestion
        self._mac = mac
        self._name = name
        self._ip_address: str | None = None
        self._last_activity: datetime | None = None
        self._connected = False
```